required/latex-lab/latex-lab-everymath.dtx

```diff
@@ -0,0 +1,347 @@
+%   Tracks display style (cases where \cs{everydisplay} is triggered).
+%    \begin{macrocode}
+\bool_new:N \l_@@_display_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\l_@@_env_bool}
+%   Tracks math environments which deal with grabbing themselves.
+%    \begin{macrocode}
+\bool_new:N \l_@@_env_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \subsection{Interface commands}
+%
+% \begin{macro}{\@@_process:n, \@@_process_aux:n}
+%   A no-op place-holder; the internal wrapper means that it does not need to
+%   be concerned with internals.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_process:n #1
+  {
+    \legacy_if:nF { measuring@ }
+      { \@@_process_aux:n {#1} }
+  }
+\cs_new_protected:Npn \@@_process_aux:n #1 { }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\math_processor:n}
+%   A simple installer
+%    \begin{macrocode}
+\cs_new_protected:Npn \math_processor:n #1
+  { \cs_set_protected:Npn \@@_process_aux:n ##1 {#1} }
+%    \end{macrocode}
+% \end{macro}
+%
+% \subsection{Content grabbing}
+%
+% \begin{macro}{\@@_grab_inline_dollar:w}
+%   Grab up to a single |$|, for inline math mode.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_inline_dollar:w % $
+  #1 $
+  {
+    \@@_process:n {#1} % $
+    #1 $
+  }
+%    \end{macrocode}
+% \end{macro}
+% \begin{macro}{\@@_grab_inline_cmd:w}
+%   The same but for the \LaTeX{} structure.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_inline_cmd:w % \(
+  #1 \)
+  {
+    \@@_process:n {#1} % \(
+    #1 \)
+  }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\@@_grab_dollardollar:w}
+%   And for the classical \TeX{} display structure.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_dollardollar:w % $$
+  #1 $$
+  {
+    \@@_process:n {#1} % $$
+    #1 $$
+  }
+%    \end{macrocode}
+% \end{macro}
+%
+% \subsection{Document commands}
+%
+% \begin{macro}{\(, \)}
+%   Here, we need to add the boolean around the structure; we also need to
+%   ensure that the conditional is finished before the |$| activates. To do
+%   that, we switch to an \pkg{expl3} structure for ease. There's no downside
+%   here to engine-protected, so we also simplify the setup a little.
+%    \begin{macrocode}
+\cs_gset_protected:Npn \( % \)
+  {
+    \bool_set_true:N \l_@@_cmd_bool
+    \mode_if_math:TF
+      { \@badmath }
+      { $ }
+  } % \(
+\cs_gset_protected:Npn \) %
+  {
+    \mode_if_math:TF
+      { $ }
+      { \@badmath }
+    \bool_set_false:N \l_@@_cmd_bool
+  }
+\cs_undefine:c { ( \c_space_tl }
+\cs_undefine:c { ) \c_space_tl }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}
+%   {\equation, \@@_equation_begin:, \equation*, \@@_equation_star_begin:}
+% \begin{macro}
+%   {\endequation, \@@_equation_end:, \endequation*, \@@_equation_star_end:}
+%   These environments are not set up by \pkg{amsmath} to collect their body,
+%   so we do that here. This has to be done \emph{after} we can be sure
+%   \pkg{amsmath} is loaded.
```

Comment:
that would become simpler if we require amsmath early on. On the other hand perhaps the actual loading should even then only happen late, so not sure this is true.

Reply:
I've favour forcing loading earlier but didn't want to do more than necessary

Reply:
that's ok, I was just commenting on the fact that we may want to go into that direction (but it doesn't have to be initially like that.